.github/workflows/clang-format-check.yml

```diff
@@ -0,0 +1,20 @@
+name: check-clang-format
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.18.1
+      with:
+        source: 'src test'
+        exclude: './third_party ./external'
+        extensions: 'h,cpp'
+        clangFormatVersion: 18.1.8
```

Comment:
The thing is I wasn't aware that something changed between `18.1.3` and newer versions when I formatted the code in cb999f20b6f2934ad7c94b10d2b02f6acf74aab4 and 9a1b93e4fea27e91d14c18821be2c940adf63bf4 . But thinking about it again, you're right. I'll make it easier for us and just reformat again with `18.1.3`, already tried that locally and very it's not that bad (20 files or so).

Than we can simply `apt install clang-format` and not have this `llvm.sh` script at all.

Reply:
There's even a clang-format-18 package. But that's the default anyway. While we're at it: I submitted my request to change the line length in a separate PR #839 as this might cause a lot of formatting change :see_no_evil:

Reply:
> There's even a clang-format-18 package. But that's the default anyway. While we're at it: I submitted my request to change the line length in a separate PR #839 as this might cause a lot of formatting change 🙈

Just saw that, it's fine ;) I'll handle the line length and `18.1.3` changes.